components/brave_rewards/browser/publisher_info_database.cc

```diff
@@ -1044,29 +1044,100 @@ bool PublisherInfoDatabase::MigrateV1toV2() {
     return false;
   }
 
-  if (!CreateContributionInfoTable()) {
+  const char* name = "contribution_info";
+  sql = "CREATE TABLE ";
+  sql.append(name);
+  sql.append(
+      "("
+      "publisher_id LONGVARCHAR,"
+      "probi TEXT \"0\"  NOT NULL,"
+      "date INTEGER NOT NULL,"
+      "category INTEGER NOT NULL,"
+      "month INTEGER NOT NULL,"
+      "year INTEGER NOT NULL,"
+      "CONSTRAINT fk_contribution_info_publisher_id"
+      "    FOREIGN KEY (publisher_id)"
+      "    REFERENCES publisher_info (publisher_id)"
+      "    ON DELETE CASCADE)");
+  if (!GetDB().Execute(sql.c_str())) {
+    return false;
+  }
+
+  if (!GetDB().Execute(
+      "CREATE INDEX IF NOT EXISTS contribution_info_publisher_id_index "
+      "ON contribution_info (publisher_id)")) {
     return false;
   }
 
-  if (!CreateContributionInfoIndex()) {
+  // Recurring_donation
+  const char* recurring_donation = "recurring_donation";
+  if (GetDB().DoesTableExist(recurring_donation)) {
+    sql = " DROP TABLE ";
+    sql.append(recurring_donation);
+    sql.append(" ; ");
+  }
+
+  if (!GetDB().Execute(sql.c_str())) {
     return false;
   }
 
-  if (!CreateRecurringDonationTable()) {
+  name = "recurring_donation";
+  sql = "CREATE TABLE ";
+  sql.append(name);
+  sql.append(
+      "("
+      "publisher_id LONGVARCHAR NOT NULL PRIMARY KEY UNIQUE,"
+      "amount DOUBLE DEFAULT 0 NOT NULL,"
+      "added_date INTEGER DEFAULT 0 NOT NULL,"
+      "CONSTRAINT fk_recurring_donation_publisher_id"
+      "    FOREIGN KEY (publisher_id)"
+      "    REFERENCES publisher_info (publisher_id)"
+      "    ON DELETE CASCADE)");
+  if (!GetDB().Execute(sql.c_str())) {
     return false;
   }
 
-  return CreateRecurringDonationIndex();
+  return GetDB().Execute(
+      "CREATE INDEX IF NOT EXISTS recurring_donation_publisher_id_index "
+      "ON recurring_donation (publisher_id)");
 }
 
 bool PublisherInfoDatabase::MigrateV2toV3() {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
 
-  if (!CreatePendingContributionsTable()) {
+  std::string sql;
+  const char* name = "pending_contribution";
+  // pending_contribution
+  const char* pending_contribution = "pending_contribution";
+  if (GetDB().DoesTableExist(pending_contribution)) {
```

Comment:
why table drop?

Reply:
From creation script. Pending was new table to v3.

Reply:
https://github.com/brave/brave-core/commit/1d94bd41d220651ac3197318133832c4d72e3762#diff-4bf8d740c793a0cf750bc637b07f65a0R680